lib/spack/spack/relocate.py

```diff
@@ -415,6 +428,40 @@ def make_binary_placeholder(cur_path_names, allow_root):
         tty.die("Placeholder not implemented for %s" % platform.system())
 
 
+def make_link_placeholder(cur_path_names, cur_dir, old_dir):
+    """
+    Replace old install path with placeholder in absolute links
+    """
+    cur_dir = os.path.realpath(cur_dir)
+    for cur_path in cur_path_names:
+        # realpath is safe, previously ensured absolute
```

Comment:
I think this comment should be removed or expanded on (i.e. explain why is using realpath on relative symlinks unsafe). As is, the docstring for this method already mentions it takes absolute symlinks (although it could rephrase to make this more explicit).

This can be avoided by using `os.readlink`

Reply:
os.readlink only follows the link by one step, as opposed to following all links until a real package is reached. Since we put the workdir in spaces that are symlinks on some systems, we cannot use os.readlink here. That was my first thought as well.

Reply:
Previous versions of this didn't work with `os.readlink`, but now that I've improved the code around it, it works.

`os.readlink` and `os.path.realpath` aren't really equivalent though, because `os.readlink` only follows the link one step, but `os.path.realpath` follows links until it reaches a real file. In this case I think the former is preferable and I've made the change.